pyqtgraph/parametertree/parameterTypes.py

```diff
@@ -741,3 +743,240 @@ class TextParameter(Parameter):
+
+class FileDialogItem(ParameterItem):
+
+    def __init__(self, param, depth):
+        ParameterItem.__init__(self, param, depth)
+
+        self.dialogTypes={
+        "getExistingDirectory" : (QtGui.QFileDialog().getExistingDirectory,"Open directory"),
+        #"getExistingDirectoryUrl" : (QtGui.QFileDialog().getExistingDirectoryUrl,"Open directory"),
+        "getOpenFileName" : (QtGui.QFileDialog().getOpenFileName,"Open file"),
+        "getOpenFileNames" : (QtGui.QFileDialog().getOpenFileNames,"Open files"),
+        #"getOpenFileUrl" : (QtGui.QFileDialog().getOpenFileUrl,"Open directory"),
+        #"getOpenFileUrls" : (QtGui.QFileDialog().getOpenFileUrls,"Open directory"),
+        "getSaveFileName" : (QtGui.QFileDialog().getSaveFileName,"Save file"),
+        #"getSaveFileUrl" : (QtGui.QFileDialog().getSaveFileUrl,"Open directory"),
+        "openFiles" : (QtGui.QFileDialog().getOpenFileNames,"Open files"),
+        "openFile" : (QtGui.QFileDialog().getOpenFileName,"Open file"),
+        "openDirectory" : (QtGui.QFileDialog().getExistingDirectory,"Open directory"),
+        "saveFile" : (QtGui.QFileDialog().getSaveFileName,"Save file")}
+
+        self.layoutWidget = QtGui.QWidget()
+        self.layout = QtGui.QHBoxLayout()
+        self.layoutWidget.setLayout(self.layout)
+        self.button = QtGui.QPushButton(param.name())
+        self.label = QtGui.QLabel()
+        #self.layout.addSpacing(100)
+        self.layout.addWidget(self.button)
+        self.layout.addWidget(self.label)
+        self.layout.addStretch()
+        self.setText(0, '')
+
+        self.returnedValues = None
+        self.lastDirectory = os.path.dirname(os.path.abspath(__file__))
+        self.button.clicked.connect(self.buttonClicked)
+
+    def treeWidgetChanged(self):
+        ParameterItem.treeWidgetChanged(self)
+        tree = self.treeWidget()
+        if tree is None:
+            return
+
+        tree.resizeColumnToContents(0)
+        tree.setItemWidget(self, 0, self.layoutWidget)
+
+    def buttonClicked(self):
+        filterString = self.param.opts.get('filterString', "All files (*.*)")
+        dialogType = self.param.opts.get('dialogType', "getOpenFileName")
+        dialogFunction, dialogCaption = self.dialogTypes[dialogType]
+        dialogCaption = self.param.opts.get('dialogCaption', dialogCaption)
+
+        if dialogType in ["getExistingDirectory","openDirectory"]:
+            fn = dialogFunction(None, dialogCaption, self.lastDirectory)
+            fn = os.path.realpath(asUnicode(fn))
+            if fn=="": return 0
+            self.lastDirectory = os.path.dirname(fn)
+            self.label.setText(str(fn))
+            tooltip = str(os.path.realpath(fn))
+        elif dialogType in ["getOpenFileNames","openFiles"]:
+            fn = dialogFunction(None, dialogCaption, self.lastDirectory, filterString)
+            fn = [os.path.realpath(asUnicode(f)) for f in fn if os.path.isfile(f) and f != ""]
```

Comment:
this throws an Exception if the user clicks on Cancel in the dialog

Reply:
I had more difficulty porting this over without problems, so I replaced it with a file parameter I've been using for about a 8 months without many issues. It supports almost all these features, but with (IMO) a bit smoother of a option interface.

Reply:
ok, but you need to change the enums to the long spelling and also the exec_().
You should try running on PyQt6 6.1.1.

Reply:
Jeez, installed pyside6 and I see what you mean. The enums should be fixed there as well.